Add a lot of links to regex reference from regex guide, add example explanations (#33974)
mdn/content: files/en-us/web/javascript/guide/regular_expressions/assertions/index.md

console.log(fruitsStartsWithNotA); // [ 'Watermelon', 'Orange', 'Strawberry' ]
```

See more examples in the [input boundary assertion](/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Input_boundary_assertion) reference.

### Matching a word boundary

> In this example, we get fruit names that has a word that ends with "en" or "ed".

Reviewer: ```suggestion
In this example, we match fruit names containing words ending in "en" or "ed".
```
